db/typed/prefix/bucket_test.go

```diff
@@ -266,43 +308,74 @@ func runReadTests(
 			blockNumber, map1 := randomEntry(t, content)
 			hash, map2 := randomEntry(t, map1)
 			slot, _ := randomEntry(t, map2)
-			test3LayerScan(t, database, content, blockNumber, hash, slot)
+			test3LayerScan(t, database, content, false, blockNumber, hash, slot)
 		}
 	})
 }
 
-func TestPrefixedBucket(t *testing.T) {
-	database := memory.New()
-	content := make(entryMap)
+func createNewDatabase(t *testing.T) (db.KeyValueStore, entryMap) {
+	t.Helper()
+	database, err := pebble.New(t.TempDir())
+	require.NoError(t, err)
+	return database, make(entryMap)
+}
+
+func populateDatabase(
+	t *testing.T,
+	database db.KeyValueWriter,
+	content entryMap,
+	layerKeyCount int,
+) {
+	t.Helper()
+	for range layerKeyCount {
+		blockNumber := rand.Uint64()
+		content[blockNumber] = make(map[felt.Address]map[string]testEntry, layerKeyCount)
 
-	t.Run("Populate database", func(t *testing.T) {
 		for range layerKeyCount {
-			blockNumber := rand.Uint64()
-			content[blockNumber] = make(map[felt.Address]map[string]testEntry)
+			address := felt.Random[felt.Address]()
+			content[blockNumber][address] = make(map[string]testEntry, layerKeyCount)
 
 			for range layerKeyCount {
-				address := felt.Random[felt.Address]()
-				content[blockNumber][address] = make(map[string]testEntry)
-
-				for range layerKeyCount {
-					slot := cryptorand.Text()
-					value := felt.Random[felt.ClassHash]()
-
-					key := testKey{
-						number:  blockNumber,
-						address: address,
-						slot:    []byte(slot),
-					}
-					entry := testEntry{
-						key:   key,
-						value: value,
-					}
+				slot := cryptorand.Text()
+				value := felt.Random[felt.ClassHash]()
 
-					content[blockNumber][address][slot] = entry
-					require.NoError(t, bucket.Put(database, key, &value))
+				key := testKey{
+					number:  blockNumber,
+					address: address,
+					slot:    []byte(slot),
 				}
+				entry := testEntry{
+					key:   key,
+					value: value,
+				}
+
+				content[blockNumber][address][slot] = entry
+				require.NoError(t, bucket.Put(database, key, &value))
 			}
 		}
+	}
+}
+
+func insertInvalidValue(t *testing.T, database db.KeyValueWriter) testKey {
+	maxFeltBigInt := new(big.Int).Sub(fp.Modulus(), new(big.Int).SetUint64(1))
+	maxFelt := new(felt.Felt).SetBigInt(maxFeltBigInt)
```

Comment:
Nitpick: this operation is similar to `felt.Zero minus felt.One`. I think it is easier to read since it doesn't require access to `fp` or `big` libraries

```golang
maxFlet := new(felt.Felt).Sub(felt.Zero, felt.One)
```